spacy/pipeline/spancat.py

```diff
@@ -215,12 +357,24 @@ def __init__(
             "spans_key": spans_key,
             "threshold": threshold,
             "max_positive": max_positive,
+            "negative_weight": negative_weight,
+            "allow_overlap": allow_overlap,
         }
         self.vocab = vocab
         self.suggester = suggester
         self.model = model
         self.name = name
         self.scorer = scorer
+        self.add_negative_label = add_negative_label
+        if allow_overlap is None:
+            self.cfg["allow_overlap"] = True
+        elif not allow_overlap and max_positive is not None and max_positive > 1:
+            self.cfg["allow_overlap"] = True
+            msg.warn(
+                "'allow_overlap' can only be False when max_positive=1, "
+                f"but found 'max_positive': {max_positive} "
+                "SpanCategorizer is automatically configured with allow_overlap=True."
```

Comment:
Should we `raise` and quit instead?

Reply:
Maybe an error would be better for now. We can always start with an error and if someone can make a good case for why they need it, then we can consider changing it.

Since this is in a component, it shouldn't be with `msg` but with python warnings or errors.